packaging/specifiers.py

```diff
@@ -386,9 +386,12 @@ def _compare_equal(self, prospective: Version, spec: str) -> bool:
         if spec.endswith(".*"):
             # In the case of prefix matching we want to ignore local segment.
             prospective = Version(prospective.public)
+            # Convert our spec string into a Version ignoring the trailing .*
+            # This allows to get the normalized version string
+            spec_version = Version(spec[:-2])
             # Split the spec out by dots, and pretend that there is an implicit
             # dot in between a release segment and a pre-release segment.
-            split_spec = _version_split(spec[:-2])  # Remove the trailing .*
+            split_spec = _version_split(str(spec_version))
```

Comment:
Does `normalize_version` work here? It’d be much cheaper than constructing a Version object and converting it immediately back to str.

Reply:
Sorry for the late reply, I was away on vacation with limited time to answer and bad internet access.

> Does normalize_version work here? It’d be much cheaper than constructing a Version object and converting it immediately back to str.

`canonicalize_version` works with `strip_trailing_zero=False`. However, I doubt it'll be cheaper given it does the same thing: construct a Version object and builds a string similar to `Version.__str__` with added conditional statements on `strip_trailing_zero` and version argument type check.

I can understand that it's clearer with `canonicalize_version` and can go this way if clarity beats speed cost here.

Reply:
Clarity (almost) always beats speed. 🙂